src/backend/distributed/executor/multi_utility.c

```diff
@@ -3193,6 +3218,77 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 }
 
 
+/*
+ * SetSearchPathToCurrentSearchPathCommand generates a command which can
+ * set the search path to the exact same search path that the issueing node
+ * has.
+ *
+ * If the current search path is null (or doesn't have any valid schemas),
+ * the function returns NULL.
+ */
+static char *
+SetSearchPathToCurrentSearchPathCommand(void)
+{
+	StringInfo setCommand = NULL;
+	char *currentSearchPath = CurrentSearchPath();
+
+	if (currentSearchPath == NULL)
+	{
+		return NULL;
+	}
+
+	setCommand = makeStringInfo();
+	appendStringInfo(setCommand, "SET search_path TO %s;", currentSearchPath);
+
+	return setCommand->data;
+}
+
+
+/*
+ * CurrentSearchPath is a C interface for calling current_schemas(bool) that
+ * PostgreSQL exports.
+ *
+ * CurrentSchemas returns all the schemas in the seach_path that are seperated
+ * with comma (,) sign. The returned string can be used to set the search_path.
+ *
+ * The function omits implicit schemas.
+ *
+ * The function returns NULL if there are no valid schemas in the search_path,
+ * mimicing current_schemas(false) function.
+ */
+static char *
+CurrentSearchPath(void)
+{
+	StringInfo currentSearchPath = makeStringInfo();
+	List *searchPathList = fetch_search_path(false);
+	ListCell *searchPathCell;
+	bool schemaAdded = false;
+
+	foreach(searchPathCell, searchPathList)
+	{
+		char *schemaName = get_namespace_name(lfirst_oid(searchPathCell));
+
+		/* watch out for deleted namespace */
+		if (schemaName)
+		{
+			if (schemaAdded)
+			{
+				appendStringInfoString(currentSearchPath, ",");
+				schemaAdded = false;
```

Comment:
Just a very minor note: I think `schemaAdded = false;`is not necessary here and it confused me a bit, tbh.

Reply:
I agree that it is a little bit confusing. The name should probably have been `addCommaSeperator`.

But, I think it is necessary to have that variable  because we cannot always get a schema in every iteration. So, how could we reset the variable otherwise?

Reply:
The thing is, even if we set it to false here, it is immediately set back to true just outside of the curly bracket. I think the check is needed only for the first iteration, don't you think so?

Reply:
Ah, I see. Yes, that seems correct.